### ISSUE TITLE
Duplicate file name errors

### ISSUE THREAD

**Issue by brycejlowe** [ignacionelson/ProjectSend#572]
When performing a client upload, if a file has the same name as one already in the database the system crashes.

I found an error between lines 155 and 170 in upload-process-form.php, when the file is already in the database (the code does an in_array of the $file['file'] element in the $url_db_files), the $original_filename variable is never populated, so it is null.  The variable $original_filename (which is null in certain cases and shouldn't be) is then added to the $add_arguments array.  

On line 232 of upload-process-form.php, the upload_add_to_database method is called and the $add_arguments['file_original'] is null, so the post_file property is set to empty on line 155.  This causes the select statement on line 204 of file-upload.php to return now rows and the file_id property is never set.  The on line 240 of upload-process-form.php the upload_add_assignment method is called, which assumes file_id property of the PSend_Upload_File class is set, but it isn't so it attempts to add a row into the file_relations table with a null file_id, which the schema doesn't allow, causing the page to crash.

**Comment by ignacionelson** [comment]
Great find!!!! Thank you for the detailed report.
I'm currently working on replacing the whole file naming routine, but will apply a temp fix for this.

**Comment by brycejlowe** [comment]
I have created a patch for this particular error and have put a fix in a pull request.

I took out the check to see if the file is already in the database.  I do realize that is only one way of fixing the problem, however I felt it was the safest.

Take the following test case that I was using.  I have 2 files called "Contract Law.docx", they both contain **different** information.  One version contains the text "You owe me $1,000,000" and the other is "I owe you $1,000,000".  The code seems to want to replace update the "Contract Law.docx" which depending on the order you upload and who you are could be problematic (depending on who you are it could be very beneficial or not).  My fix makes it so that the title is duplicated instead, giving you some idea that the file is different.  A longer term fix, if you do want single instance storage, would be to do some type of hash of the file, check if it matches and then call them the same.
